src/sources/prometheus/scrape.rs

```diff
@@ -531,8 +529,11 @@ mod test {
         }
     }
 
+    // Intentially not using assert_source_compliance here because this is a round-trip test which
```

Comment:
Do we have other tests for these components that do assert compliance?

Reply:
Yep! All the other unit tests and integration tests in this file do.